packages/echo-base/src/errors/BaseError.test.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseError, getAllProperties } from './BaseError';
+
+describe('BaseError', () => {
+    const message = 'this is a test';
+    it('should initialize with correct values', () => {
+        const actualError = new BaseError({ name: 'BaseError', message });
+        expect(actualError.message).toBe(message);
```

Comment:
Just to help readability: one extra line before expect would help separate the setup and execution a bit.